packages/compass-schema-validation/src/components/sample-documents/sample-documents.jsx

```diff
@@ -63,9 +90,13 @@ class SampleDocuments extends Component {
    * @returns {Boolean} If the component should update.
    */
   shouldComponentUpdate(nextProps) {
+    const { sampleDocuments: prevDocs } = this.props;
+    const { sampleDocuments: nextDocs } = nextProps;
     return (
-      nextProps.sampleDocuments.isLoading !==
-      this.props.sampleDocuments.isLoading
+      prevDocs.validDocumentState !== nextDocs.validDocumentState ||
+      prevDocs.invalidDocumentState !== nextDocs.invalidDocumentState ||
+      prevDocs.validDocument !== nextDocs.validDocument ||
+      prevDocs.invalidDocument !== nextDocs.invalidDocument
```

Comment:
That's really good that you are fixing this, but at this point does it make sense to even do this `should` check? Generally speaking based on the state flow, if document `state` changed, everything else probably changed too and so this feel very redundant, maybe we can just drop this as unnecessary?

Reply:
You're right, but keeping in mind what you mentioned above regarding directly independently components, I will not even need it.

Reply:
Yeah, I guess my point is we probably don't need it either way 😄